scapy/fields.py

```diff
@@ -1557,7 +1557,10 @@ def __getattr__(self, attr):
             return all(bool((2 ** self.names.index(flag)) & int(self))
                        for flag in attr)
         except ValueError:
-            return super(FlagValue, self).__getattr__(attr)
+            try:
+                return self.__getattr__(attr.replace('_', '-'))
```

Comment:
Why not (after all, I suspect we have very little chances to have a `-` vs `_` name collision). But I would do the other way around: first, try with unmodified `attr` value, and if it fails, try again with the modified value.

Reply:
That is why I tried to write. How do you want to change it?

Reply:
You're very right, I was probably too tired.

Reply:
Isn't there a possibility that we enter an infinite recursion here?

What happens if I ask for a flag that does not exist? To prevent that, we might
```Python
        except ValueError:
            if '_' in attr:
                try:
                    return self.__getattr__(attr.replace('_', '-'))
                except AttributError:
                    pass
            return super(FlagValue, self).__getattr__(attr)
 ```

What do you think?

Reply:
Of course, if we have a flag value with both a `_` and a `-`, we're screwed. But I guess this fix doesn't have to be perfect. BTW, maybe a comment wouldn't hurt here!

Reply:
Thanks. I modified the PR.

I believe that we should not allow '-' in the field names.

Reply:
@guedou agreed. `-` should be forbidden in field and flag names...